web-registry/src/features/marketplace/BuySellOrderFlow/BuySellOrderFlow.tsx

```diff
@@ -107,23 +97,95 @@ export const BuySellOrderFlow = ({
     error,
     setError,
   } = useMsgClient(handleTxQueued, handleTxDelivered, handleError);
+
   const accountAddress = wallet?.address;
   const txHash = deliverTxResponse?.transactionHash;
   const txHashUrl = getHashUrl(txHash);
   const errorEnum = findErrorByCodeEnum({ errorCode: error });
   const ErrorIcon = errorsMapping[errorEnum].icon;
 
-  const project = useMemo(
-    () => ({
-      id: selectedProject?.id.toString() ?? '',
-      sellOrders: projectUiSellOrdersInfo?.filter(
+  /**
+   * data processing (business logic)
+   */
+  const [selectedProject, setSelectedProject] =
+    useState<ProjectWithOrderData>();
+
+  useEffect(() => {
+    if (projects?.length === 1) setSelectedProject(projects[0]);
+  }, [projects]);
```

Comment:
This might be better as a hook for naming:
ie: `useAutoSelectFirstProject()`

Reply:
@flagrede instead I have extracted the selectedProject state and logic to a `useSelectedProject` custom hook, also encapsulating the function that makes the selection from the projectId, `setSelectedProjectById`